Add missing `@covers` annotations

pbearne opened the pull request:
the ongoing covers ticket 
I am still rechecking all the files so expect up[date to the pull request

Trac:
https://core.trac.wordpress.org/ticket/56782
https://core.trac.wordpress.org/ticket/39265

pbearne committed:
Merge pull request #1 from WordPress/master

catch

pbearne committed:
Actions folder

pbearne committed:
Root tests

pbearne committed:
Fixed Root tests

pbearne committed:
Fixed Root tests part 2
Resolves

pbearne committed:
Fixed Root tests part 2
Resolves

pbearne committed:
The A folders

pbearne committed:
The Block folder

pbearne committed:
The bookmark folder

pbearne committed:
Remove not needed/user return var

pbearne committed:
Fixed setup/tearDown

pbearne committed:
Merge pull request #2 from WordPress/master

update

pbearne committed:
the C folders

pbearne committed:
Merge branch 'master' into #39265_covers

# Conflicts:
#	tests/phpunit/tests/blocks/register.php

pbearne committed:
Work todate

pbearne committed:
Whitespaces and bad test fixed

pbearne committed:
renamed @cover ::autop to ::wpautop

pbearne committed:
removed @cover per function to use the global

pbearne committed:
removed @cover per function to use the global

pbearne committed:
removed @cover per function to use the global

pbearne committed:
checked F folders

pbearne committed:
checked g folder

pbearne committed:
checked h folder

pbearne committed:
checked i folders

pbearne committed:
checked L folders

pbearne committed:
Merge branch 'WordPress:master' into master

pbearne committed:
Merge branch 'master' into #39265_covers

# Conflicts:
#	tests/phpunit/tests/blocks/block-editor.php
#	tests/phpunit/tests/post/objects.php

pbearne committed:
checked L folders

pbearne committed:
Merge branch 'WordPress:master' into master

pbearne committed:
Merge branch 'master' into #39265_covers

pbearne committed:
checked M folders

pbearne committed:
checked O folders

pbearne committed:
checked P folders

pbearne committed:
checked Q folders

hellofromtonya commented on tests/phpunit/tests/formatting/ExcerptRemoveBlocks.php:
```diff
@@ -22,7 +22,7 @@ class Tests_Formatting_ExcerptRemoveBlocks extends WP_UnitTestCase {
 	<div class="wp-block-column">
 		<!-- wp:archives {"displayAsDropdown":false,"showPostCounts":false} /-->
-		
+
```
@pbearne this change and the one on line 43 are impacting the failing test. Plus it's not needed as part of covers. Reverting the change here and on line 43 should resolve the failing test.

hellofromtonya commented on tests/phpunit/tests/formatting/ExcerptRemoveBlocks.php:
```diff
@@ -40,7 +40,7 @@ class Tests_Formatting_ExcerptRemoveBlocks extends WP_UnitTestCase {
 
 		<p>paragraph inside column</p>
-		
+
```
@pbearne Same here. Revert this change.

pbearne committed:
reverted IDE striped tab

pbearne committed:
checked S and T folders

pbearne committed:
Move test from editPost.php

pbearne committed:
the last files
All that is left is the rest API folder

pbearne committed:
Merge branch 'WordPress:master' into master

pbearne committed:
Merge branch 'master' into #39265_covers

jrfnl commented:
@pbearne Awesome work!

Some questions:

* Have all remarks which I left on the previous PR #1221 been addressed ?
* Why are there changes to the grunt setup included in this PR ?
* Should the `forceCoversAnnotation="true"` attribute be added to the PHPUnit configuration files at this time ?
    Note: there are two config files - the `phpunit.xml.dist` file in the project root and the `multisite.xml` file in the `tests/phpunit` directory. Both would need to be adjusted.

spacedmonkey committed:
Add covers to REST API endpoints

pbearne committed:
reverted changes to Grunfile.js and package.json

pbearne committed:
Merge branch 'WordPress:master' into master

pbearne committed:
Merge branch 'master' into #39265_covers

# Conflicts:
#	tests/phpunit/tests/theme/wpThemeJson.php

pbearne committed:
Merge pull request #3 from spacedmonkey/fix/covers

Add covers to REST API endpoints props @spacedmonkey

pbearne commented:
> @pbearne Awesome work!
> 
> Some questions:
> 
> * Have all remarks which I left on the previous PR #1221 been addressed ?
Yes all done
> * Why are there changes to the grunt setup included in this PR ?
Reverted not sure how they got in there
> * Should the `forceCoversAnnotation="true"` attribute be added to the PHPUnit configuration files at this time ?
Would be nice I am trying to do the rest_api folder and will set the flag once done
>   Note: there are two config files - the `phpunit.xml.dist` file in the project root and the `multisite.xml` file in the `tests/phpunit` directory. Both would need to be adjusted.

Time for many eyes lets get this done

Paul

pbearne committed:
reverted changes to themeDir.php

pbearne committed:
Merge branch 'WordPress:master' into #39265_covers

pbearne committed:
sett forceCoversAnnotation to true

pbearne committed:
Merge branch 'WordPress:master' into master

pbearne committed:
Merge branch 'WordPress:master' into #39265_covers

pbearne committed:
added missing covers

pbearne committed:
Merge pull request #4 from WordPress/master

Docs: Correct DocBlock formatting for some revision tests.

hellofromtonya commented on tests/phpunit/tests/actions.php:
```diff
@@ -309,6 +350,9 @@ function test_action_keyed_array() {
 	}
 
+	/**
+	 * @covers ::do_action
```
```suggestion
	 * @covers ::remove_action
```

This test intends to test a one-shot circuit, i.e. a circuit where a callback can only be invoked one time. It's testing the registration process. How so? `do_action` loops through the registered callbacks. It will only invoke those callbacks that are registered to the event (tag) being fired.

In this case, the function being tested is the `remove_action` _when_ the callback gets invoked the first time. The callback could be directly invoked to test the functionality. But here it's using `do_action` as the invoking mechanism.

Note: 
This test needs work as It's not actually doing the test it intends to do. Why? `do_action` is called only once meaning the callback will only be called once. It needs to fire again to ensure the callback does not get invoked a second time. The test will need to be fixed in another ticket and patch. Adding to my list.

hellofromtonya commented on tests/phpunit/tests/actions.php:
```diff
@@ -348,6 +397,8 @@ function action_that_causes_recursion( $tag ) {
 	 * @ticket 9968
 	 * @ticket 17817
+	 *
+	 * @covers ::add_action
```
```suggestion
	 * @covers ::remove_action
	 * @covers ::add_action
```

This test is testing the callback registration (add) and deregistration (remove) process. Reviewing the tickets attached it, it's checking for a specific use case of when the callback registration is dynamically changed when the event (tag) is being invoked.

hellofromtonya commented on tests/phpunit/tests/actions.php:
```diff
@@ -416,6 +469,8 @@ function test_remove_anonymous_callback() {
 	 *
 	 * @ticket 17817
+	 *
+	 * @covers ::has_action
```
```suggestion
 	 * @covers WP_Hook::offsetGet
	 * @covers WP_Hook::offsetSet
 	 * @covers WP_Hook::offsetUnset
```

This test is testing `ArrayAccess`, i.e. being able to use the instance of `WP_Hook` (which is stored in the global `$wp_filter`) as an array.

hellofromtonya commented on tests/phpunit/tests/actions.php:
```diff
@@ -509,6 +572,9 @@ function test_doing_filter_real() {
 		$this->assertFalse( doing_filter( 'testing' ) ); // No longer doing this filter.
 	}
 
+	/**
+	 * @covers ::doing_filter
+	 */
```
```suggestion
```

This is a callback method used by test `test_doing_filter_real`.

hellofromtonya commented on tests/phpunit/tests/actions.php:
```diff
@@ -530,6 +596,9 @@ function apply_testing_filter() {
 		$this->assertFalse( doing_filter( 'testing_nested' ) );
 	}
 
+	/**
+	 * @covers ::doing_filter
+	 */
```
```suggestion
```
This is a callback method used by test `test_doing_filter_real`. The `@covers` should be applied to the test itself.

hellofromtonya reviewed (changes requested):
Great job @pbearne. Here's a review of the `tests/phpunit/tests/actions.php` file along with explanations. 

For `test_action_priority`, it's testing how priorities are handled when registering (adding) a callback to an event. This is handled in `WP_Hook::add_filter`. For this specific test, it's checking the priority of an action callback using `add_action`. So this tests `@covers` is `add_action`.

pbearne committed:
Update tests/phpunit/tests/actions.php

Co-authored-by: Tonya Mork <tonya.mork@automattic.com>

pbearne committed:
Update tests/phpunit/tests/actions.php

Co-authored-by: Tonya Mork <tonya.mork@automattic.com>

pbearne committed:
Update tests/phpunit/tests/actions.php

Co-authored-by: Tonya Mork <tonya.mork@automattic.com>

pbearne committed:
Update tests/phpunit/tests/actions.php

Co-authored-by: Tonya Mork <tonya.mork@automattic.com>

pbearne committed:
Update tests/phpunit/tests/actions.php

Co-authored-by: Tonya Mork <tonya.mork@automattic.com>

pbearne committed:
Update tests/phpunit/tests/actions.php

Co-authored-by: Tonya Mork <tonya.mork@automattic.com>

pbearne committed:
Update tests/phpunit/tests/actions.php

Co-authored-by: Tonya Mork <tonya.mork@automattic.com>

pbearne committed:
Update tests/phpunit/tests/actions.php

Co-authored-by: Tonya Mork <tonya.mork@automattic.com>

pbearne committed:
Merge branch 'WordPress:master' into master

pbearne committed:
Merge branch 'master' into #39265_covers

# Conflicts:
#	tests/phpunit/tests/bookmark/getBookmarks.php
#	tests/phpunit/tests/widgets.php

pbearne commented:
Updated from master and accepted the code review changes
I totally expect that correction/change will be made to the covers overtime I really just what to 98%+ correct and all in :-)

hellofromtonya commented on tests/phpunit/tests/actions/callbacks.php:
```diff
@@ -7,6 +7,8 @@ class Tests_Actions_Callbacks extends WP_UnitTestCase {
 	/**
 	 * @ticket 23265
+	 *
+	 * @covers ::has_action
```
```suggestion
	 * @covers ::add_action
```

Ticket 23265 improved `_wp_filter_build_unique_id` to make it easier for `remove_action` and other hook functions to work with callbacks in different representations. It's part of the callback registration process. Notice that it's also a private function to encapsulate reusable functionality for use within core. As it's private, the covers is applied to the consuming function, i.e. `add_action` in this case.

hellofromtonya commented on tests/phpunit/tests/actions/closures.php:
```diff
@@ -9,6 +9,9 @@ class Tests_Actions_Closures extends WP_UnitTestCase {
 	/**
 	 * @ticket 10493
+	 *
+	 * @covers ::has_action
+	 * @covers ::do_action
```
This test is testing multiple things:

- `add_action`: registering a closure as the callback
- `has_action`: checking the closure was successfully registered as the callback
- `do_action`: closure was successfully invoked when the action event fired

It does cover all of these currently.

```suggestion
	 * @covers ::add_action
	 * @covers ::has_action
	 * @covers ::do_action
```

Note: This test is doing too much and should be split into separate tests (in another ticket).

hellofromtonya reviewed (changes requested):
Hey @pbearne this review covers `tests/phpunit/tests/actions/callbacks.php` and `tests/phpunit/tests/actions/closures.php`.

pbearne committed:
Update tests/phpunit/tests/actions/callbacks.php

Co-authored-by: Tonya Mork <tonya.mork@automattic.com>

pbearne committed:
Update tests/phpunit/tests/actions/closures.php

Co-authored-by: Tonya Mork <tonya.mork@automattic.com>

pbearne committed:
Merge branch 'master' into #39265_covers

pbearne committed:
There were 39 warnings:

1) Tests_Admin_includesMisc::test_shorten_url
Trying to @cover or @use not existing method "::url_shorten;".

2) Tests_Admin_includesTemplate::test_add_meta_box
Trying to @cover or @use not existing method "::".

3) Tests_Admin_includesTemplate::test_add_meta_box_on_multiple_screens
Trying to @cover or @use not existing method "::".

4) Tests_Admin_includesTemplate::test_add_meta_box_with_previously_removed_box_and_sorted_priority
Trying to @cover or @use not existing method "::".

5) WP_Test_Block_Editor::test_block_editor_context_no_settings
Trying to @cover or @use not existing method "::WP_Block_Editor_Context".

6) WP_Test_Block_Editor::test_block_editor_context_post
Trying to @cover or @use not existing method "::WP_Block_Editor_Context".

7) WP_Test_Block_Register::test_unregister_affects_main_registry
Trying to @cover or @use not existing method "::register_block_type     ".

8) Tests_Comment_Walker::test_has_children
Trying to @cover or @use not existing method "Walker_Comment::__construct".

9) Tests_Compat::test_json_encode_decode
Trying to @cover or @use not existing method "Services_JSON::encodeUnsafe".

10) Tests_WP_Customize_Manager::test_fresh_site_flag_clearing
Trying to @cover or @use not existing method "::_publish_changeset_values".

11) Test_WP_Customize_Selective_Refresh::test_register_scripts
Trying to @cover or @use not existing method "WP_Customize_Selective_Refresh::register_scripts".

12) Tests_WP_Customize_Widgets::test_wp_widget_form_customize_control_json
Trying to @cover or @use not existing method "WP_Customize_Widgets::json".

13) Tests_Media::test_get_image_send_to_editor_defaults
Trying to @cover or @use not existing class or interface ":get_image_send_to_editor:".

14) Tests_Option_SiteOption::test_update_adds_falsey_value
Trying to @cover or @use not existing method "::flush_cache".

15) Tests_POMO_MO::test_mo_simple
Trying to @cover or @use not existing method "MO::headers".

16) Tests_POMO_MO::test_mo_context
Trying to @cover or @use not existing method "Translation_Entry::context".

17) Tests_POMO_TranslationEntry::test_create_entry
Trying to @cover or @use not existing method "Translation_Entry::singular".

18) Tests_POMO_Translations::test_add_entry
Trying to @cover or @use not existing method "Translation_Entry::add_entry".

19) Tests_POMO_Translations::test_translate
Trying to @cover or @use not existing method "Translation_Entry::translate".

20) Tests_POMO_Translations::test_translate_plural
Trying to @cover or @use not existing method "Translation_Entry::translate_plural".

21) Tests_POMO_Translations::test_digit_and_merge
Trying to @cover or @use not existing method "Translation_Entry::translate".

22) Tests_Post_Nav_Menu::test_wp_update_nav_menu_item_with_special_characters_in_category_name
Trying to @cover or @use not existing method "::wp_get_nav_menu_item".

23) Tests_Rewrite::test_add_rule
Trying to @cover or @use not existing method "::add_rule".

24) Tests_Rewrite::test_add_rule_redirect_array
Trying to @cover or @use not existing method "::add_rule".

25) Tests_Rewrite::test_add_rule_top
Trying to @cover or @use not existing method "::add_rule".

26) Tests_Rewrite::test_flush_rules_does_not_delete_option
Trying to @cover or @use not existing method "::flush_rules".

27) Test_Sitemaps::test_get_url_list_page_with_home
Trying to @cover or @use not existing method "::wp_get_sitemap_providers.".

28) Test_Sitemaps::test_sitemaps_enabled
Trying to @cover or @use not existing class or interface "WP_Sitemaps:sitemaps_enabled".

29) Test_Sitemaps::test_disable_sitemap_should_return_404
Trying to @cover or @use not existing class or interface "WP_Sitemaps:render_sitemaps".

30) Test_Sitemaps::test_empty_url_list_should_return_404
Trying to @cover or @use not existing class or interface "WP_Sitemaps:render_sitemaps".

31) Tests_Theme::test_default_themes_have_textdomain
Trying to @cover or @use not existing method "WP_Theme::wp_get_theme".

32) Tests_Theme::test_year_in_readme
Trying to @cover or @use not existing method "WP_Theme::wp_get_theme".

33) Tests_Theme_ThemeDir::test_get_theme_data_top_level
Trying to @cover or @use not existing method "::get_themes_data".

34) Tests_Theme_ThemeDir::test_get_theme_data_subdir
Trying to @cover or @use not existing method "::get_themes_data".

35) Tests_User::test_user_properties
Trying to @cover or @use not existing method "WP_User::data".

36) Tests_User_Capabilities::test_user_level_update
Trying to @cover or @use not existing method "WP_Role::set_role".

37) Tests_User_Capabilities::test_current_user_edit_caps
Trying to @cover or @use not existing method "::add_cap".

38) Tests_User_Capabilities::test_get_roles_data
Trying to @cover or @use not existing method "WP_Roles::roles".

39) Tests_User_WpDeleteUser::test_get_blogs_of_user
Trying to @cover or @use not existing method "::wpmu_delete_user".

pbearne committed:
Merge remote-tracking branch 'origin/#39265_covers' into #39265_covers

pbearne committed:
1) Tests_Admin_includesMisc::test_shorten_url
Trying to @cover or @use not existing method "::url_shorten;".

2) WP_Test_Block_Editor::test_block_editor_context_post
Trying to @cover or @use not existing method "::WP_Block_Editor_Context".

3) Tests_Compat::test_json_encode_decode
Trying to @cover or @use not existing method "Services_JSON::encodeUnsafe".

4) Test_Sitemaps::test_get_url_list_page_with_home
Trying to @cover or @use not existing method "WP_Sitemaps_Provider::get_url_list.".

5) Tests_User::test_user_properties
Trying to @cover or @use not existing method "WP_User::$data".

6) Tests_User_Capabilities::test_get_roles_data
Trying to @cover or @use not existing method "WP_Roles::$roles".

pbearne committed:
replace space with tabs for cs2pr tests

hellofromtonya commented on tests/phpunit/tests/admin/includesListTable.php:
```diff
@@ -75,6 +75,8 @@ public static function wpSetUpBeforeClass( WP_UnitTest_Factory $factory ) {
 	/**
 	 * @ticket 15459
+	 *
+	 * @covers WP_Posts_List_Table::display_rows
```
```suggestion
	 * @covers WP_Posts_List_Table::display_rows
	 * @covers WP_Posts_List_Table::set_hierarchical_display
```
This one is testing that "hierarchical pages" are rendered correctly from the query results. It's indirectly testing the query results but directly testing the list table's building of the HTML.

hellofromtonya commented on tests/phpunit/tests/admin/includesListTable.php:
```diff
@@ -91,6 +93,8 @@ function test_list_hierarchical_pages_first_page() {
 	/**
 	 * @ticket 15459
+	 *
+	 * @covers WP_Posts_List_Table::display_rows
```
```suggestion
	 * @covers WP_Posts_List_Table::display_rows
	 * @covers WP_Posts_List_Table::set_hierarchical_display
```
Same here.

hellofromtonya commented on tests/phpunit/tests/admin/includesListTable.php:
```diff
@@ -108,6 +112,8 @@ function test_list_hierarchical_pages_second_page() {
 	/**
 	 * @ticket 15459
+	 *
+	 * @covers WP_Posts_List_Table::display_rows
```
```suggestion
	 * @covers WP_Posts_List_Table::display_rows
	 * @covers WP_Posts_List_Table::set_hierarchical_display
```
Same here.

hellofromtonya commented on tests/phpunit/tests/admin/includesListTable.php:
```diff
@@ -125,6 +131,8 @@ function test_search_hierarchical_pages_first_page() {
 	/**
 	 * @ticket 15459
+	 *
+	 * @covers WP_Posts_List_Table::display_rows
```
```suggestion
	 * @covers WP_Posts_List_Table::display_rows
	 * @covers WP_Posts_List_Table::set_hierarchical_display
```
Same here.

hellofromtonya commented on tests/phpunit/tests/admin/includesListTable.php:
```diff
@@ -142,6 +150,8 @@ function test_search_hierarchical_pages_second_page() {
 	/**
 	 * @ticket 15459
+	 *
+	 * @covers WP_Posts_List_Table::display_rows
```
```suggestion
	 * @covers WP_Posts_List_Table::display_rows
	 * @covers WP_Posts_List_Table::set_hierarchical_display
```
Same here.

hellofromtonya commented on tests/phpunit/tests/admin/includesListTable.php:
```diff
@@ -161,6 +171,8 @@ function test_grandchildren_hierarchical_pages_first_page() {
 	/**
 	 * @ticket 15459
+	 *
+	 * @covers WP_Posts_List_Table::display_rows
```
```suggestion
	 * @covers WP_Posts_List_Table::display_rows
	 * @covers WP_Posts_List_Table::set_hierarchical_display
```
Same here.

hellofromtonya commented on tests/phpunit/tests/admin/includesListTable.php:
```diff
@@ -389,6 +419,8 @@ function() {
 	/**
 	 * @ticket 45089
+	 *
+	 * @covers WP_Posts_List_Table::get_sortable_columns
```
```suggestion
	 * @covers WP_Posts_List_Table::print_column_headers
```
The test isn't actually running the  code in`get_sortable_columns` but rather is mocking it and ensuring that it is invoked. What is being tested? That `print_column_headers` renders columns that are sortable where "sortable" is a condition. Also noting that `get_sortable_columns` is a protected method used to configure the columns being rendered.

hellofromtonya commented on tests/phpunit/tests/admin/includesListTable.php:
```diff
@@ -423,6 +455,8 @@ public function test_sortable_columns() {
 	/**
 	 * @ticket 45089
+	 *
+	 * @covers WP_Posts_List_Table::get_sortable_columns
```
```suggestion
	 * @covers WP_Posts_List_Table::print_column_headers
```
Same here.

hellofromtonya reviewed (changes requested):
This round of review covers (pun intended 🤣 ) these files:

- tests/phpunit/tests/admin/includesComment.php
- tests/phpunit/tests/admin/includesCommunityEvents.php
- tests/phpunit/tests/admin/includesFile.php
- tests/phpunit/tests/admin/includesListTable.php
- tests/phpunit/tests/admin/includesMisc.php

pbearne committed:
Update tests/phpunit/tests/admin/includesCommunityEvents.php

Co-authored-by: Tonya Mork <tonya.mork@automattic.com>

pbearne committed:
Update tests/phpunit/tests/admin/includesListTable.php

Co-authored-by: Tonya Mork <tonya.mork@automattic.com>

pbearne committed:
Update tests/phpunit/tests/admin/includesListTable.php

Co-authored-by: Tonya Mork <tonya.mork@automattic.com>

pbearne committed:
Update tests/phpunit/tests/admin/includesListTable.php

Co-authored-by: Tonya Mork <tonya.mork@automattic.com>

pbearne committed:
Update tests/phpunit/tests/admin/includesListTable.php

Co-authored-by: Tonya Mork <tonya.mork@automattic.com>

pbearne committed:
Update tests/phpunit/tests/admin/includesListTable.php

Co-authored-by: Tonya Mork <tonya.mork@automattic.com>

pbearne committed:
Update tests/phpunit/tests/admin/includesListTable.php

Co-authored-by: Tonya Mork <tonya.mork@automattic.com>

pbearne committed:
Update tests/phpunit/tests/admin/includesListTable.php

Co-authored-by: Tonya Mork <tonya.mork@automattic.com>

pbearne committed:
Update tests/phpunit/tests/admin/includesListTable.php

Co-authored-by: Tonya Mork <tonya.mork@automattic.com>

pbearne committed:
fixed covers as per hellofromtonya change

pbearne committed:
Merge branch 'WordPress:master' into #39265_covers

hellofromtonya commented on tests/phpunit/tests/admin/includesPost.php:
```diff
@@ -130,6 +136,8 @@ function test__wp_translate_postdata_cap_checks_editor() {
 	 *
 	 * @ticket 25272
+	 *
+	 * @covers ::wp_insert_post
```
```suggestion
	 * @covers ::edit_post
```

Technically you're right, the change happens in `wp_insert_post` as it sets the default value when not passed to it. However, `edit_post` is the function that's setting up the post data and then invokes `wp_insert_post`. In this case, `post_status` is not set in the post data, which then gets set in `wp_insert_post` as the default of `draft`. 

So this test is specifically testing what happens when invoking `edit_post` without passing the `post_status` parameter it. This makes this test an integration test.

hellofromtonya commented on tests/phpunit/tests/admin/includesSchema.php:
```diff
@@ -226,6 +230,8 @@ public function data_populate_site_meta() {
 	 * @group multisite
 	 * @dataProvider data_populate_network_meta
+	 *
+	 * @covers ::populate_site_meta
```
```suggestion
	 * @covers ::populate_network_meta
```

hellofromtonya commented on tests/phpunit/tests/admin/includesScreen.php:
```diff
@@ -302,6 +320,9 @@ function test_post_type_edit_collisions() {
 	}
 
+	/**
+	 * @covers WP_Screen::add_help_tab
```
```suggestion
	 * @covers WP_Screen::add_help_tab
	 * @covers WP_Screen:: get_help_tab
	 * @covers WP_Screen:: get_help_tabs
	 * @covers WP_Screen::remove_help_tab
	 * @covers WP_Screen::remove_help_tabs
```

This test is testing all of the help tab(s) methods (doing too much).

hellofromtonya commented on tests/phpunit/tests/admin/includesScreen.php:
```diff
@@ -336,6 +357,10 @@ function test_help_tabs() {
 	/**
 	 * @ticket 19828
+	 *
+	 * @covers WP_Screen::add_help_tab
+	 * @covers WP_Screen::get_help_tab
+	 * @covers WP_Screen::remove_help_tab
```
```suggestion
	 * @covers WP_Screen::add_help_tab
	 * @covers WP_Screen:: get_help_tab
	 * @covers WP_Screen:: get_help_tabs
	 * @covers WP_Screen::remove_help_tab
	 * @covers WP_Screen::remove_help_tabs
```

This test is testing all of the help tab(s) methods (doing too much) but adds testing the order as set by the priority.

hellofromtonya commented on tests/phpunit/tests/admin/includesScreen.php:
```diff
@@ -431,6 +456,9 @@ function test_help_tabs_priority() {
 	/**
 	 * @ticket 25799
+	 *
+	 * @covers WP_Screen::add_option
+	 * @covers WP_Screen::get_option
```
```suggestion
	 * @covers WP_Screen::add_option
	 * @covers WP_Screen::get_option
	 * @covers WP_Screen::remove_option
	 * @covers WP_Screen::remove_options
```

This one is testing all of the option methods. (doing too much)

hellofromtonya commented on tests/phpunit/tests/admin/includesScreen.php:
```diff
@@ -606,6 +637,9 @@ public function data_is_block_editor() {
 	 * }
 	 * @param bool   $expected The expected `is_block_editor` value.
+	 *
+	 * @covers ::set_current_screen
+	 * @covers ::get_current_screen
```
```suggestion
	 * @covers ::set_current_screen
```

The work to setup the block editor properties is done when running `set_current_screen` as it invokes `get` static method which handles the setup. There is a method called `is_block_editor()` which seems more like the test subject of this specific test. It's currently not though. So without using that method, this test is validating the initial instantiation is done correctly when the current screen is for the block editor.

hellofromtonya reviewed (changes requested):
This code review includes these files: 

- tests/phpunit/tests/admin/includesPlugin.php
- tests/phpunit/tests/admin/includesPost.php
- tests/phpunit/tests/admin/includesSchema.php
- tests/phpunit/tests/admin/includesScreen.php

pbearne committed:
Update tests/phpunit/tests/admin/includesPlugin.php

Co-authored-by: Tonya Mork <tonya.mork@automattic.com>

pbearne committed:
Update tests/phpunit/tests/admin/includesPlugin.php

Co-authored-by: Tonya Mork <tonya.mork@automattic.com>

pbearne committed:
Update tests/phpunit/tests/admin/includesSchema.php

Co-authored-by: Tonya Mork <tonya.mork@automattic.com>

pbearne committed:
Update tests/phpunit/tests/admin/includesScreen.php

Co-authored-by: Tonya Mork <tonya.mork@automattic.com>

pbearne committed:
Update tests/phpunit/tests/admin/includesPost.php

Co-authored-by: Tonya Mork <tonya.mork@automattic.com>

pbearne committed:
fixed white space

spacedmonkey commented on tests/phpunit/tests/rest-api/rest-block-type-controller.php:
```diff
@@ -1,4 +1,4 @@
-<?php
+§<?php
```
```suggestion
<?php
```

spacedmonkey commented on tests/phpunit/tests/rest-api/rest-widget-types-controller.php:
```diff
@@ -14,7 +14,7 @@
  * @see WP_TEST_REST_Controller_Testcase
  * @group restapi
- * @covers WP_REST_Widget_Types_Controller
+ * @coversDefaultClass WP_REST_Widget_Types_Controller
```
@hellofromtonya We need to make sure we correct all of these.

spacedmonkey commented on tests/phpunit/tests/multisite/networkQuery.php:
```diff
@@ -8,6 +8,8 @@
 	 * @group ms-network-query
 	 * @group multisite
+	 *
+	 * @covers WP_Network_Query::query
```
```suggestion
	 * @coversDefaultClass WP_Network_Query
```

spacedmonkey replied:
The methods should have `@covers ::query`

pbearne replied:
as all test are the same function I thick do a single @covers on the class is clearer 
if we had different functions from the same class then your change would work

spacedmonkey commented on tests/phpunit/tests/multisite/siteQuery.php:
```diff
@@ -114,6 +114,9 @@ public static function wpTearDownAfterClass() {
 		}
 
+		/**
+		 * @covers WP_Site_Query::query
```
Again here, we should use @coversDefaultClass on the class and @covers on the method.

spacedmonkey commented on tests/phpunit/tests/oembed/controller.php:
```diff
@@ -279,6 +296,10 @@ function test_request_without_url_param() {
 	}
 
+	/**
+	 * @covers WP_REST_Server::dispatch
+	 * @covers WP_HTTP_Requests_Response::get_data
```
that is not what this covers. It covers this https://github.com/WordPress/wordpress-develop/blob/4b7b089896c966f61585db121242a1c79de425fa/src/wp-includes/class-wp-oembed-controller.php#L118-L138

pbearne replied:
all of these need someone who understands the underlining function that are been tested

pbearne committed:
Merge branch 'WordPress:master' into #39265_covers

pbearne committed:
Update tests/phpunit/tests/admin/includesScreen.php

Co-authored-by: Tonya Mork <tonya.mork@automattic.com>

pbearne committed:
Update tests/phpunit/tests/admin/includesScreen.php

Co-authored-by: Tonya Mork <tonya.mork@automattic.com>

pbearne committed:
fixed test_is_block_editor comment

pbearne committed:
Update tests/phpunit/tests/rest-api/rest-block-type-controller.php

Co-authored-by: Jonny Harris <spacedmonkey@users.noreply.github.com>

pbearne committed:
Merge branch 'WordPress:master' into master

pbearne committed:
Merge branch 'WordPress:master' into #39265_covers

pbearne committed:
Merge branch 'WordPress:master' into master

pbearne committed:
Merge branch 'master' into #39265_covers

# Conflicts:
#	tests/phpunit/tests/blocks/render-reusable.php
#	tests/phpunit/tests/rest-api/rest-sidebars-controller.php
#	tests/phpunit/tests/theme/wpTheme.php
#	tests/phpunit/tests/theme/wpThemeJson.php
#	tests/phpunit/tests/xmlrpc/wp/editPost.php

pbearne committed:
Merge branch 'master' into #39265_covers

# Conflicts:
#	phpunit.xml.dist
#	tests/phpunit/multisite.xml
#	tests/phpunit/tests/admin/includesMisc.php
#	tests/phpunit/tests/admin/wpCommunityEvents.php
#	tests/phpunit/tests/ajax/Attachments.php
#	tests/phpunit/tests/ajax/MediaEdit.php
#	tests/phpunit/tests/comment/getCommentsPagesCount.php
#	tests/phpunit/tests/compat.php
#	tests/phpunit/tests/formatting/escJs.php
#	tests/phpunit/tests/formatting/redirect.php
#	tests/phpunit/tests/formatting/urlShorten.php
#	tests/phpunit/tests/formatting/wpHtmlExcerpt.php
#	tests/phpunit/tests/formatting/wpSpecialchars.php
#	tests/phpunit/tests/formatting/wpTexturize.php
#	tests/phpunit/tests/functions/wpListFilter.php
#	tests/phpunit/tests/functions/wpListUtil.php
#	tests/phpunit/tests/image/editorGd.php
#	tests/phpunit/tests/media.php
#	tests/phpunit/tests/multisite/msFilesRewriting.php
#	tests/phpunit/tests/option/option.php
#	tests/phpunit/tests/option/themeMods.php
#	tests/phpunit/tests/rest-api/rest-attachments-controller.php
#	tests/phpunit/tests/sitemaps/functions.php
#	tests/phpunit/tests/sitemaps/wpSitemapsRegistry.php
#	tests/phpunit/tests/theme/wpThemeJsonResolver.php
#	tests/phpunit/tests/xmlrpc/wp/getMediaItem.php
#	tests/phpunit/tests/xmlrpc/wp/uploadFile.php

pbearne committed:
Merge branch 'master' into #39265_covers

pbearne committed:
end of day save

pbearne committed:
added update code

pbearne committed:
Merge branch 'trunk' into #39265_covers

# Conflicts:
#	tests/phpunit/tests/admin/includesListTable.php
#	tests/phpunit/tests/admin/includesMisc.php
#	tests/phpunit/tests/admin/includesPlugin.php
#	tests/phpunit/tests/admin/includesPost.php
#	tests/phpunit/tests/admin/includesScreen.php
#	tests/phpunit/tests/auth.php
#	tests/phpunit/tests/basic.php
#	tests/phpunit/tests/blocks/serialize.php
#	tests/phpunit/tests/cache.php
#	tests/phpunit/tests/canonical/paged.php
#	tests/phpunit/tests/comment/commentForm.php
#	tests/phpunit/tests/comment/query.php
#	tests/phpunit/tests/comment/template.php
#	tests/phpunit/tests/cron.php
#	tests/phpunit/tests/customize/setting.php
#	tests/phpunit/tests/db.php
#	tests/phpunit/tests/dependencies.php
#	tests/phpunit/tests/file.php
#	tests/phpunit/tests/filesystem/base.php
#	tests/phpunit/tests/filesystem/findFolder.php
#	tests/phpunit/tests/filters.php
#	tests/phpunit/tests/formatting/balanceTags.php
#	tests/phpunit/tests/formatting/cleanPre.php
#	tests/phpunit/tests/formatting/convertInvalidEntries.php
#	tests/phpunit/tests/formatting/escAttr.php
#	tests/phpunit/tests/formatting/escHtml.php
#	tests/phpunit/tests/formatting/escUrl.php
#	tests/phpunit/tests/formatting/isEmail.php
#	tests/phpunit/tests/formatting/makeClickable.php
#	tests/phpunit/tests/formatting/redirect.php
#	tests/phpunit/tests/formatting/sanitizeFileName.php
#	tests/phpunit/tests/formatting/sanitizeTitle.php
#	tests/phpunit/tests/formatting/sanitizeUser.php
#	tests/phpunit/tests/formatting/slashit.php
#	tests/phpunit/tests/formatting/wpHtmlExcerpt.php
#	tests/phpunit/tests/formatting/wpTexturize.php
#	tests/phpunit/tests/formatting/zeroise.php
#	tests/phpunit/tests/functions.php
#	tests/phpunit/tests/general/wpGetDocumentTitle.php
#	tests/phpunit/tests/image/dimensions.php
#	tests/phpunit/tests/image/functions.php
#	tests/phpunit/tests/image/header.php
#	tests/phpunit/tests/image/siteIcon.php
#	tests/phpunit/tests/image/size.php
#	tests/phpunit/tests/import/parser.php
#	tests/phpunit/tests/includes/factory.php
#	tests/phpunit/tests/kses.php
#	tests/phpunit/tests/l10n/loadTextdomain.php
#	tests/phpunit/tests/link.php
#	tests/phpunit/tests/mail.php
#	tests/phpunit/tests/media.php
#	tests/phpunit/tests/meta.php
#	tests/phpunit/tests/multisite.php
#	tests/phpunit/tests/multisite/msFilesRewriting.php
#	tests/phpunit/tests/multisite/network.php
#	tests/phpunit/tests/multisite/site.php
#	tests/phpunit/tests/oembed/controller.php
#	tests/phpunit/tests/oembed/discovery.php
#	tests/phpunit/tests/oembed/filterResult.php
#	tests/phpunit/tests/oembed/getResponseData.php
#	tests/phpunit/tests/oembed/postEmbedUrl.php
#	tests/phpunit/tests/oembed/template.php
#	tests/phpunit/tests/option/multisite.php
#	tests/phpunit/tests/option/option.php
#	tests/phpunit/tests/option/siteOption.php
#	tests/phpunit/tests/option/siteTransient.php
#	tests/phpunit/tests/option/themeMods.php
#	tests/phpunit/tests/option/transient.php
#	tests/phpunit/tests/option/userSettings.php
#	tests/phpunit/tests/option/wpLoadAllOptions.php
#	tests/phpunit/tests/pomo/mo.php
#	tests/phpunit/tests/pomo/noopTranslations.php
#	tests/phpunit/tests/pomo/po.php
#	tests/phpunit/tests/pomo/translationEntry.php
#	tests/phpunit/tests/pomo/translations.php
#	tests/phpunit/tests/post.php
#	tests/phpunit/tests/post/attachments.php
#	tests/phpunit/tests/post/filtering.php
#	tests/phpunit/tests/post/formats.php
#	tests/phpunit/tests/post/getPages.php
#	tests/phpunit/tests/post/meta.php
#	tests/phpunit/tests/post/nav-menu.php
#	tests/phpunit/tests/post/objects.php
#	tests/phpunit/tests/post/query.php
#	tests/phpunit/tests/post/template.php
#	tests/phpunit/tests/post/thumbnails.php
#	tests/phpunit/tests/post/types.php
#	tests/phpunit/tests/post/wpGetPostParentId.php
#	tests/phpunit/tests/post/wpListPages.php
#	tests/phpunit/tests/rest-api.php
#	tests/phpunit/tests/rest-api/rest-users-controller.php
#	tests/phpunit/tests/rewrite.php
#	tests/phpunit/tests/shortcode.php
#	tests/phpunit/tests/taxonomy.php
#	tests/phpunit/tests/term.php
#	tests/phpunit/tests/term/getTermLink.php
#	tests/phpunit/tests/theme.php
#	tests/phpunit/tests/theme/customHeader.php
#	tests/phpunit/tests/theme/support.php
#	tests/phpunit/tests/theme/wpThemeJson.php
#	tests/phpunit/tests/theme/wpThemeJsonResolver.php
#	tests/phpunit/tests/upload.php
#	tests/phpunit/tests/url.php
#	tests/phpunit/tests/user.php
#	tests/phpunit/tests/user/author.php
#	tests/phpunit/tests/user/capabilities.php
#	tests/phpunit/tests/user/multisite.php
#	tests/phpunit/tests/user/query.php
#	tests/phpunit/tests/user/session.php
#	tests/phpunit/tests/user/wpDeleteUser.php
#	tests/phpunit/tests/walker.php
#	tests/phpunit/tests/widgets.php
#	tests/phpunit/tests/xmlrpc/basic.php

pbearne committed:
Merge pull request #10 from WordPress/trunk

catch up

pbearne committed:
Merge branch 'WordPress:trunk' into #39265_covers

pbearne committed:
Merge branch 'WordPress:trunk' into #39265_covers

pbearne committed:
work so far

pbearne committed:
work so far

pbearne committed:
work so far

pbearne committed:
work so far

pbearne committed:
seems to be working

pbearne committed:
seems to be working

pbearne committed:
Merge branch 'trunk' into #39265_covers

# Conflicts:
#	tests/phpunit/tests/admin/includesPlugin.php

pbearne committed:
Merge branch 'WordPress:trunk' into #39265_covers

pbearne committed:
Merge branch 'WordPress:trunk' into #39265_covers

pbearne committed:
Merge branch 'WordPress:trunk' into #39265_covers

pbearne committed:
Merge branch 'trunk' into #39265_covers

# Conflicts:
#	tests/phpunit/tests/cache.php
#	tests/phpunit/tests/theme/support.php

pbearne committed:
Merge branch 'WordPress:trunk' into #39265_covers

pbearne committed:
Merge branch 'trunk' into #39265_covers

# Conflicts:
#	tests/phpunit/tests/theme/support.php
#	tests/phpunit/tests/user/capabilities.php

pbearne committed:
Merge remote-tracking branch 'origin/#39265_covers' into #39265_covers

ironprogrammer committed:
Change test class name to better match intent

ironprogrammer committed:
Replace with `@coversNothing`

Existing `@covers` give a false impression for intent of tests. The functions in question are tested elsewhere in a direct, intentional manner.

This update mitigates false positive coverage.

ironprogrammer committed:
Rename and move file to match path in source

Since these tests are for `/readme.html`, move to more consistent, intuitive location.

ironprogrammer committed:
Add description to test class

ironprogrammer committed:
Remove redundant function signature

ironprogrammer committed:
Fix function declarations

Replaces `Tests_Formatting_BalanceTagsfunction`.

ironprogrammer committed:
Migrate setUp/tearDown functions

Resolves phpunit error:

PHP Fatal error:  Declaration of Tests_Image_Editor_GD::setUp() must be compatible with Yoast\PHPUnitPolyfills\TestCases\TestCase::setUp(): void

ironprogrammer committed:
Update covers for cron tests

pbearne committed:
Merge branch 'trunk' into #39265_covers

# Conflicts:
#	tests/phpunit/tests/dependencies/scripts.php
#	tests/phpunit/tests/query/stickies.php

pbearne committed:
Merge branch 'trunk' into #39265_covers

# Conflicts:
#	tests/phpunit/tests/multisite/network.php

pbearne committed:
Merge branch 'WordPress:trunk' into #39265_covers

pbearne committed:
Merge branch 'trunk' into #39265_covers

# Conflicts:
#	tests/phpunit/tests/post/types.php
#	tests/phpunit/tests/rest-api/rest-server.php

anton-vlasenko committed:
Update @covers tag for Tests_Ajax_AddMeta class

anton-vlasenko committed:
Add @covers tag for Tests_Ajax_AddTag class.

anton-vlasenko committed:
Add @covers tag for Tests_Ajax_Attachments class.

anton-vlasenko committed:
Fixed @covers tag for Tests_Ajax_CompressionTest::test_logged_out.
@coversNothing method is not needed here because test method is testing if the error is thrown.

anton-vlasenko committed:
Add @covers tag for Tests_Ajax_CustomizeMenus class.

anton-vlasenko committed:
Add @covers tag for Tests_Ajax_DeleteComment class.

anton-vlasenko committed:
Add @covers tag for Tests_Ajax_Delete_Plugin::test_delete_plugin class.

anton-vlasenko committed:
Fix @covers tag for Tests_Ajax_Manage_Themes class.

Add @covers tag for Tests_Ajax_QuickEdit class.

anton-vlasenko committed:
Refactor Tests_Attachment_Slashes class.
Move @covers tag to the class level PHPDOC block to save some space.

anton-vlasenko committed:
Refactor Test_Block_Supports_Border class.
Refactor Tests_Block_Supports_Colors class.
Refactor Tests_Block_Supports_Elements class.
Refactor Test_Block_Supports_Layout class.
Refactor Test_Block_Supports_Spacing class.
Refactor Tests_Block_Supports_Typography class.
Move @covers tag to the class level PHPDOC block to save some space.

anton-vlasenko committed:
Add additional @covers tags for Tests_L10n_LoadTextdomain class.
Add additional @covers tags for Tests_L10n_LoadTextdomainJustInTime class.
Add additional @covers tags for Tests_L10n_TranslateSettingsUsingI18nSchema class.
Add additional @covers tags for Tests_L10n_wpLocaleSwitcher class.
Add additional @covers tags for Tests_L10n_wpLocaleSwitcher class.

anton-vlasenko committed:
Remove unused variable in the Tests_Multisite_Option::test_users_can_register_signup_filter.
Add additional @covers tags for Tests_Option_NetworkOption class.
Add additional @covers tags for Tests_Option_Registration class.
Add additional @covers tags for Tests_Option_Theme_Mods class.
Refactor Tests_Option_Theme_Mods::test_theme_mod_set_with_invalid_theme_mods_option. There is no need to call test_theme_mod_set_with_invalid_theme_mods_option method statically.
Add additional @covers tags for Tests_Option_Transient class.
Fix typo in Tests_Option_UpdateOption::test_autoload_should_not_be_updated_for_existing_option_when_value_is_changed_but_no_value_of_autoload_is_provided.
Add additional @covers tags for Tests_User_Settings class.

anton-vlasenko committed:
Add additional @covers tags for Tests_Url_GetPrivacyPolicyUrl class.

anton-vlasenko committed:
Add additional @covers tags for Tests_URL class.

ironprogrammer committed:
Checkout file from trunk

Required to fix unit tests.

Version of this file in the PR was old and had a bug introduced after WPCS 1.0 update (I did say old). Unable to find history of this bug in trunk (??) so did a checkout[overwrite] to ensure this file is up to date.

ironprogrammer committed:
Add `@covers` to Tests_Formatting_ConvertInvalidEntities

ironprogrammer committed:
Rename file corrects `Entries` -> `Entities`

Filename now matches covered test class (`Tests_Formatting_ConvertInvalidEntities`) and function name (`convert_invalid_entities`).

ironprogrammer committed:
Update `@covers` for Tests_Formatting_EscUrl

ironprogrammer committed:
Remove class-level `@covers`

Tests in class cover two different functions.

ironprogrammer committed:
Add test `@covers` in Tests_Formatting_GetBloginfo

Test includes checking output from `sanitize_option()`.

ironprogrammer committed:
Remove redundant `@covers` in Tests_Formatting_SanitizeOrderby

ironprogrammer committed:
Remove redundant `@covers` from DocBlock

ironprogrammer committed:
Shift `@ticket` notation to more consistent order

ironprogrammer committed:
Consolidate `@covers` to class-level

`::wpautop` is the only function covered in this class.

ironprogrammer committed:
Add missing `@covers` for KSES tests

ironprogrammer committed:
Correct `@covers` reference to tested function

pbearne committed:
Merge branch 'trunk' into #39265_covers

# Conflicts:
#	tests/phpunit/tests/admin/includesPost.php
#	tests/phpunit/tests/admin/includesTemplate.php
#	tests/phpunit/tests/image/functions.php
#	tests/phpunit/tests/kses.php
#	tests/phpunit/tests/media.php
#	tests/phpunit/tests/pluggable.php
#	tests/phpunit/tests/privacy/wpPrivacyGeneratePersonalDataExportFile.php
#	tests/phpunit/tests/rest-api/rest-attachments-controller.php
#	tests/phpunit/tests/rewrite/oldDateRedirect.php
#	tests/phpunit/tests/rewrite/oldSlugRedirect.php
#	tests/phpunit/tests/sitemaps/wpSitemapsPosts.php
#	tests/phpunit/tests/user/mapMetaCap.php
#	tests/phpunit/tests/user/wpListAuthors.php

anton-vlasenko committed:
Add @covers tags for Tests_Block_Template_Utils class.

Add @covers tags for Tests_Block_Template class.

Add @covers tags for Tests_Blocks_Context and Tests_Blocks_Editor classes.

Add @covers tags for Tests_Blocks_Register class.

Add @covers tags in Tests_Blocks_RenderReusableCommentTemplate class.

Add @covers tags for Tests_Blocks_SupportedStyles and Tests_Blocks_wpBlock classes.

Add @covers tags for Tests_Blocks_wpBlockType class.

anton-vlasenko committed:
Remove redundant @covers tags from Tests_Bookmark_GetBookmark class.

Remove redundant @covers tags from Tests_Bookmark_GetBookmarks class.

anton-vlasenko committed:
Revert "Remove redundant @covers tags from Tests_Bookmark_GetBookmark class."

This reverts commit 06e4718fd3b43c072050d377622ddb73d8780d99.

anton-vlasenko committed:
Fix code review feedback.
Please see https://github.com/pbearne/wordpress-develop/commit/06b6bc6d46d433d7253d2323c1a95dd984c372d9#r76369062 and https://github.com/pbearne/wordpress-develop/commit/73bdcee655ee083a17c58cce8be1f6773223b512#r76368720

anton-vlasenko committed:
Remove redundant @covers tags from Tests_Canonical_HTTPS class.

Add @covers tags for Tests_Canonical_PostStatus class.

Add @covers tags for Tests_Canonical_Sitemaps class.

anton-vlasenko committed:
Add missing @covers tags for Tests_Canonical class.

anton-vlasenko committed:
Add missing @covers tags for Tests_Comment class.
Add missing @covers tags for Tests_Comment_Submission class.
Remove duplicate @covers tag in Tests_Comment_CommentForm.
Add missing @covers tag in Tests_Comment_Meta_Cache.
Add missing @covers tag in Tests_Comment_Query.

anton-vlasenko committed:
Fix @covers tags in Tests_DB_Charset and Tests_Cache.
Fix @covers tags in Tests_DB class.
Fix @covers tags in Tests_dbDelta class.

anton-vlasenko committed:
Remove redundant @covers annotations in Tests_Editor_wpEditors class.

anton-vlasenko committed:
Add @covers tag in Tests_Feed_Atom class.

anton-vlasenko committed:
Fix @covers tag in Tests_Feed_RSS2 class.

anton-vlasenko committed:
Fix @covers tag in Tests_Error_Protection_wpRecoveryModeCookieService and Tests_Error_Protection_wpRecoveryModeKeyService class.

anton-vlasenko committed:
Fix @covers tag in TestFactoryFor and Tests_TestHelpers class.

anton-vlasenko committed:
Fix @covers tag in Tests_Dependencies class.

pbearne committed:
Merge branch 'trunk' into #39265_covers

# Conflicts:
#	tests/phpunit/tests/db.php
#	tests/phpunit/tests/formatting/escUrl.php

anton-vlasenko committed:
Fix @covers tag in Tests_Import_Import class.

anton-vlasenko committed:
Fix @covers tag in Tests_Import_Postmeta class.

anton-vlasenko committed:
Fix code review feedback. More info here: https://github.com/pbearne/wordpress-develop/commit/be107c28823e96bc8b6ad8a749a55e82d7de9d7b

anton-vlasenko committed:
Remove duplicate @covers tags from Tests_Url_GetPrivacyPolicyUrl class.

anton-vlasenko committed:
Fix missing @covers tags in Tests_File class.
Move the @covers tag to the class level.
Fix @covers tags in WP_Filesystem_UnitTestCase class.
WP_Filesystem_Find_Folder_Test covers a mock class. Mock classes are not supposed to be added to the coverage report.

anton-vlasenko committed:
Fix @covers tags in Tests_Dependencies_jQuery.
Fix @covers tags in Tests_Dependencies_Scripts.
Fix @covers tags in Tests_Dependencies_Styles.
Fix @covers tags in Tests_Functions_wpInlineScriptTag.

pbearne committed:
stashing

anton-vlasenko committed:
Fix @covers tags in Test_WP_Customize_Control class.
Fix @covers tags in Tests_WP_Customize_Manager class.
Fix @covers tags in Test_WP_Customize_Nav_Menu_Item_Setting.
Fix @covers tags in Test_WP_Customize_Nav_Menu_Setting.
Fix @covers tags in Test_WP_Customize_Nav_Menus.
Fix @covers tags in Test_WP_Customize_Selective_Refresh.
Fix @covers tags in Tests_WP_Customize_Setting.
Fix @covers tags in Tests_WP_Customize_Widgets.

pbearne committed:
got it work

anton-vlasenko committed:
Remove redundant @covers tags in Tests_Date_CurrentTime.
Remove redundant @covers tags in Tests_Date_DateI18n.
Remove redundant @covers tags in Tests_Date_GetCommentDate.
Remove redundant @covers tags in Tests_Date_GetFeedBuildDate.
Remove redundant @covers tags in Tests_Date_GetPermalink.
Remove redundant @covers tags in WP_UnitTestCase.
Remove redundant @covers tags in Tests_Date_GetTheDate.
Remove redundant @covers tags in Tests_Date_GetTheDate.
Remove redundant @covers tags in Tests_Date_MaybeDeclineDate.
Remove redundant @covers tags in Tests_Date_mysql2date.
Fix @covers tags in Tests_Date_Query.
Fix @covers tags in Tests_Date_TheDate.
Remove redundant @covers tags.
Remove redundant @covers tags in Tests_Date_wpTimezone.
Fix @covers tags in Tests_Date_XMLRPC.

anton-vlasenko committed:
Fix @covers tags in Tests_Functions.
Fix @covers tags in Tests_Functions_Anonymization
Fix @covers tags in Tests_Functions_CanonicalCharset.
Remove redundant @covers tags in Tests_Functions_Deprecated.
Remove redundant @covers tags in Tests_Functions_DoEnclose.

anton-vlasenko committed:
In my opinion code coverage report should only cover actual WordPress classes/functions and not the test suite code.
For this reason, I'd use @coversNothing.

anton-vlasenko committed:
Fix code review feedback. Please see https://github.com/pbearne/wordpress-develop/commit/2520583f5f39ef86027f62ff0680a50741d41e9f#r77973097 and https://github.com/pbearne/wordpress-develop/commit/2520583f5f39ef86027f62ff0680a50741d41e9f#commitcomment-77973145.
These methods seem to be more appropriate for @covers.

anton-vlasenko committed:
Fix code review feedback. Please see https://github.com/pbearne/wordpress-develop/commit/b0516379f6ea1cabb252898b45ff4880cd9a7b77#commitcomment-77971068

anton-vlasenko committed:
Fix code review feedback. https://github.com/pbearne/wordpress-develop/commit/ec999980a71c56688a9061663d69f5e2390c8aee#commitcomment-77968033

anton-vlasenko committed:
Fix code review feedback. Please see https://github.com/pbearne/wordpress-develop/commit/ccc18c7d1cf399f4143c7936a65b6638c0298743#r77967343

anton-vlasenko committed:
Fix code review feedback. Please see https://github.com/pbearne/wordpress-develop/commit/903bb5380bd6bd0c5700e3355e4d730bb3b83b1b#r77957507 and https://github.com/pbearne/wordpress-develop/commit/903bb5380bd6bd0c5700e3355e4d730bb3b83b1b#r77959033.

anton-vlasenko committed:
Fix code review feedback. See https://github.com/pbearne/wordpress-develop/commit/be107c28823e96bc8b6ad8a749a55e82d7de9d7b#r79581818

anton-vlasenko committed:
Add missing @covers tags in Tests_Shortcode.
Add missing @covers tags in Tests_Site_Health.
Fix "@covers" tags in Tests_General_wpError.

pbearne committed:
added test for cache

pbearne committed:
added test for cache

anton-vlasenko committed:
Fix @covers tags in TestFactoryFor and Tests_TestHelpers.

pbearne committed:
FIxed the conect not returning the same key for first and later calls

pbearne committed:
whitespace

pbearne committed:
Merge branch 'trunk' into #39265_covers

# Conflicts:
#	tests/phpunit/tests/admin/includesPost.php
#	tests/phpunit/tests/basic.php
#	tests/phpunit/tests/block-template-utils.php
#	tests/phpunit/tests/blocks/register.php
#	tests/phpunit/tests/cache.php
#	tests/phpunit/tests/comment/query.php
#	tests/phpunit/tests/date/query.php
#	tests/phpunit/tests/dependencies/scripts.php
#	tests/phpunit/tests/filters.php
#	tests/phpunit/tests/functions.php
#	tests/phpunit/tests/hooks/addFilter.php
#	tests/phpunit/tests/l10n/loadTextdomainJustInTime.php
#	tests/phpunit/tests/option/multisite.php
#	tests/phpunit/tests/option/networkOption.php
#	tests/phpunit/tests/option/option.php
#	tests/phpunit/tests/post.php
#	tests/phpunit/tests/rest-api/rest-widgets-controller.php
#	tests/phpunit/tests/site-health.php
#	tests/phpunit/tests/taxonomy.php
#	tests/phpunit/tests/term/getEditTermLink.php
#	tests/phpunit/tests/theme/customHeader.php
#	tests/phpunit/tests/url.php

pbearne committed:
Update template.php

whitespace extra space at end of line

pbearne committed:
fixed test

pbearne committed:
Merge remote-tracking branch 'origin/#39265_covers' into #39265_covers

pbearne committed:
fixed whitespace

hellofromtonya commented on tests/phpunit/tests/dependencies/jquery.php:
```diff
@@ -87,7 +88,7 @@ public function test_dont_allow_deregister_core_scripts_in_admin() {
 	 * @ticket 28404
 	 *
-	 * @covers ::wp_script_is
+	 * @covers ::wp_enqueue_script
```
Actually, this one is intending to test `wp_script_is()` as denoted by the test method's name.

I'm relocating this test to a new test class for `wp_script_is()` and including the `@covers` for it. See https://github.com/WordPress/wordpress-develop/pull/4244.

```suggestion
	 * @covers ::wp_script_is
```

hellofromtonya committed:
Restore covers on Tests_Dependencies_jQuery::test_wp_script_is_dep_enqueued()

pbearne committed:
Merge branch 'trunk' into #39265_covers

# Conflicts:
#	tests/phpunit/tests/admin/includesTemplate.php
#	tests/phpunit/tests/admin/includesUser.php
#	tests/phpunit/tests/admin/wpPrivacyRequestsTable.php
#	tests/phpunit/tests/block-template-utils.php
#	tests/phpunit/tests/block-template.php
#	tests/phpunit/tests/blocks/register.php
#	tests/phpunit/tests/blocks/serialize.php
#	tests/phpunit/tests/comment/wpBlacklistCheck.php
#	tests/phpunit/tests/comment/wpBlocklistCheck.php
#	tests/phpunit/tests/comment/wpCheckCommentDisallowedList.php
#	tests/phpunit/tests/dependencies.php
#	tests/phpunit/tests/dependencies/scripts.php
#	tests/phpunit/tests/feed/rss2.php
#	tests/phpunit/tests/functions.php
#	tests/phpunit/tests/functions/deprecated.php
#	tests/phpunit/tests/functions/doEnclose.php
#	tests/phpunit/tests/https-detection.php
#	tests/phpunit/tests/image/editorGd.php
#	tests/phpunit/tests/image/editorImagick.php
#	tests/phpunit/tests/kses.php
#	tests/phpunit/tests/link/getThePrivacyPolicyLink.php
#	tests/phpunit/tests/media.php
#	tests/phpunit/tests/multisite/network.php
#	tests/phpunit/tests/multisite/site.php
#	tests/phpunit/tests/multisite/wpCountSites.php
#	tests/phpunit/tests/oembed/filterResult.php
#	tests/phpunit/tests/option/multisite.php
#	tests/phpunit/tests/option/networkOption.php
#	tests/phpunit/tests/pomo/pluralForms.php
#	tests/phpunit/tests/post/getPages.php
#	tests/phpunit/tests/post/nav-menu.php
#	tests/phpunit/tests/post/query.php
#	tests/phpunit/tests/post/template.php
#	tests/phpunit/tests/post/wpUniquePostSlug.php
#	tests/phpunit/tests/privacy/wpCreateUserRequest.php
#	tests/phpunit/tests/privacy/wpPrivacyCompletedRequest.php
#	tests/phpunit/tests/privacy/wpPrivacyDeleteOldExportFiles.php
#	tests/phpunit/tests/privacy/wpPrivacyGeneratePersonalDataExportFile.php
#	tests/phpunit/tests/privacy/wpPrivacyGeneratePersonalDataExportGroupHtml.php
#	tests/phpunit/tests/privacy/wpPrivacyProcessPersonalDataExportPage.php
#	tests/phpunit/tests/privacy/wpPrivacySendErasureFulfillmentNotification.php
#	tests/phpunit/tests/privacy/wpPrivacySendPersonalDataExportEmail.php
#	tests/phpunit/tests/privacy/wpPrivacySendRequestConfirmationNotification.php
#	tests/phpunit/tests/rest-api.php
#	tests/phpunit/tests/rest-api/rest-attachments-controller.php
#	tests/phpunit/tests/rest-api/rest-block-type-controller.php
#	tests/phpunit/tests/rest-api/rest-sidebars-controller.php
#	tests/phpunit/tests/rest-api/rest-widget-types-controller.php
#	tests/phpunit/tests/rest-api/rest-widgets-controller.php
#	tests/phpunit/tests/rewrite.php
#	tests/phpunit/tests/rewrite/rewriteTags.php
#	tests/phpunit/tests/shortcode.php
#	tests/phpunit/tests/sitemaps/functions.php
#	tests/phpunit/tests/sitemaps/wpSitemapsUsers.php
#	tests/phpunit/tests/term/getTheTerms.php
#	tests/phpunit/tests/term/query.php
#	tests/phpunit/tests/term/taxQuery.php
#	tests/phpunit/tests/theme/customHeader.php
#	tests/phpunit/tests/theme/wpThemeJson.php
#	tests/phpunit/tests/theme/wpThemeJsonResolver.php
#	tests/phpunit/tests/user/author.php
#	tests/phpunit/tests/user/capabilities.php
#	tests/phpunit/tests/user/query.php
#	tests/phpunit/tests/wp/addQueryVar.php

pbearne committed:
Merge branch 'WordPress:trunk' into #39265_covers

pbearne committed:
fixed bad merge

pbearne committed:
Merge remote-tracking branch 'origin/#39265_covers' into #39265_covers

pbearne committed:
fixed bad merge

pbearne committed:
fixed bad merge

pbearne committed:
fixed bad merge

pbearne committed:
Merge branch 'WordPress:trunk' into #39265_covers

spacedmonkey commented:
@pbearne Can I recommend breaking this into smaller PR. Or at least classes should be in new PR. This PR is really hard to review and harder to commit / revert.

pbearne committed:
Merge branch 'WordPress:trunk' into #39265_covers

pbearne committed:
fixed bad merge

pbearne committed:
Merge branch 'trunk' into #39265_covers

# Conflicts:
#	tests/phpunit/tests/blocks/renderBlock.php

pbearne committed:
Merge branch 'WordPress:trunk' into #39265_covers

pbearne committed:
fixed bad merge

pbearne committed:
Merge branch 'trunk' into #39265_covers

pbearne committed:
Merge branch 'trunk' into #39265_covers

# Conflicts:
#	tests/phpunit/tests/rest-api/rest-users-controller.php

pbearne committed:
Merge branch 'trunk' into #39265_covers

# Conflicts:
#	tests/phpunit/tests/block-template-utils.php
#	tests/phpunit/tests/date/getPostTime.php
#	tests/phpunit/tests/date/query.php
#	tests/phpunit/tests/date/theDate.php
#	tests/phpunit/tests/date/xmlrpc.php
#	tests/phpunit/tests/filesystem/findFolder.php
#	tests/phpunit/tests/functions/anonymization.php
#	tests/phpunit/tests/functions/canonicalCharset.php
#	tests/phpunit/tests/functions/referer.php
#	tests/phpunit/tests/link.php
#	tests/phpunit/tests/term.php
